stripe/src/main/java/com/stripe/android/view/CardNumberEditText.java

```diff
@@ -201,6 +213,10 @@ private void updateCardBrand(String partialNumber) {
         setFilters(new InputFilter[] {new InputFilter.LengthFilter(mLengthMax)});
     }
 
+    private void updateCardBrandFromNumber(String partialNumber) {
+        updateCardBrand(CardUtils.getPossibleCardType(partialNumber));
```

Comment:
I separated updating the brand via brand name and via number because there might be cause to change directly without reading the number, particularly if more testing is needed on that area, or if we need to override certain types of matches. Both methods are private (and hit by tests!)